### ISSUE TITLE
No Vary: Accept-Encoding nor cache extent when resource is not improved

### ISSUE THREAD

**Issue by Lofesa** [pagespeed/ngx_pagespeed#1323]
Hi there
I have a site on a VM, centos 7.2, nginx 1.11.6 NPS 1.11.33.4, and Wordpress moved from http to https http/2. When in http, css and js files where minified and combined with a wp plugin. I am triying to no combine nor inline these files because is the recommended with http/2, but have some issue with that:
1 css file is not url rewriten, not cache extended.... and no Vary Accept-Encoding, no Etag. When debugging I get `<!--CSS rewrite failed: Cannot improve https://example.com/wp-content/themes/theme-name/css/responsive.min.css-->`, but the file is inlined (I think because Prioritize Critical Css). 
Headers:
HTTP/1.1 200 OK
Server: nginx
Content-Type: text/css
Transfer-Encoding: chunked
Connection: keep-alive
Date: Thu, 17 Nov 2016 19:55:35 GMT
Expires: Thu, 24 Nov 2016 19:55:35 GMT
Cache-Control: max-age=604800
Last-Modified: Fri, 28 Oct 2016 09:27:27 GMT
Content-Encoding: br

1 css file is not url rewriten too, nor cache extend, but have Etag and Vary: Accept-Encoding. User-Agent, Save-Data, no debug data because is a file called from a js script and don´t appear in the html code
Headers:
HTTP/1.1 200 OK
Server: nginx
Content-Type: text/css
Transfer-Encoding: chunked
Connection: keep-alive
X-Original-Content-Length: 273557
Etag: W/"PSA-aj-iduyDQ1HIB"
Date: Thu, 17 Nov 2016 19:55:35 GMT
Expires: Thu, 24 Nov 2016 19:55:35 GMT
Cache-Control: max-age=604800
Vary: Accept-Encoding, User-Agent,Save-Data
Content-Encoding: br

Some js (allmost 9) files that have url rewriten and cached extended, but don´t have Vary: Accept-Encoded, No debug messages and the rewriten url are type file.js.pagespeed.ce.aHvzEE8Xou.js
Headers:
HTTP/1.1 200 OK
Server: nginx
Content-Type: application/javascript
Transfer-Encoding: chunked
Connection: keep-alive
Last-Modified: Fri, 28 Oct 2016 01:00:15 GMT
Date: Thu, 17 Nov 2016 19:55:34 GMT
Expires: Fri, 17 Nov 2017 19:55:34 GMT
Cache-Control: max-age=31536000
Etag: W/"0"
X-Page-Speed: 1.11.33.4-0
Content-Encoding: br

Here the filters applied:
Filters
ij	Canonicalize Javascript library URLs
cw	Collapse Whitespace
gp	Convert Gif to Png
jp	Convert Jpeg to Progressive
jw	Convert Jpeg To Webp
mc	Convert Meta Tags
pj	Convert Png to Jpeg
ws	When converting images to WebP, prefer lossless conversions
ea	Elide Attributes
ec	Cache Extend Css
ei	Cache Extend Images
ep	Cache Extend PDFs
es	Cache Extend Scripts
fc	Fallback Rewrite Css 
hw	Flushes html
gf	Inline Google Font CSS
il	Inline @import to Link
io	In-place optimize for browser
id	Insert Image Dimensions
js	Jpeg Subsampling
tu	Left Trim Urls
cj	Move Css Above Scripts
cm	Move Css To Head
pc	Add pedantic types
pr	Prioritize Critical Css
rj	Recompress Jpeg
rp	Recompress Png
rw	Recompress Webp
rc	Remove Comments
rq	Remove Quotes
ri	Resize Images
ir	Resize to Rendered Image Dimensions
cf	Rewrite Css
jm	Rewrite External Javascript
jj	Rewrite Inline Javascript
cs	Rewrite Style Attributes
cu	Rewrite Style Attributes With Url
cp	Strip Image Color Profiles
md	Strip Image Meta Data

If nedeed I can send url to the site.

**Comment by Lofesa** [comment]
Files get cache extend and url are rewrited but files that "only" are cache extended don´t have the header Vary: Accept-Encoding nor x-original-content-length. This last seem normal, if file are unchanged content-length don´t change.

**Comment by Lofesa** [comment]
Changed NPS to 1.12.34.1 and the issue is ongoing.
The unchanged resources are getting a s-maxage of 10. not Vary: Accept-Encodding not etag....

**Comment by Lofesa** [comment]
Sorry for calling you here @jeffkaufman but this issue sound related to this other in mod_pagespeed [#1382](https://github.com/pagespeed/mod_pagespeed/issues/1382)

**Comment by jeffkaufman** [comment]
Yes, they do look potentially related.  None of us have dug into this bug yet, though.
